### PR TITLE
Kindsys: Include @grafanamaturity counts to Kinds report

### PR DESCRIPTION
**Pull request by joanlopez**
**What is this feature?**

It extends the current [Kinds report (.json)](https://github.com/grafana/grafana/blob/main/pkg/kindsys/report.json) to include  the total count of how many `@grafanamaturity` occurrences has each schema.

**Why do we need this feature?**

To give more visibility of such "remaining work" to refine the already schematized entities. 

**Who is this feature for?**

Those who interact with (especially maintain) code generated types.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/schematization-and-as-code-project/issues/41

**Special notes for your reviewer**:

Note these changes are written on top of https://github.com/grafana/grafana/pull/61910.

### CODE REVIEW

File: pkg/kindsys/kindsysreport/kindsysreport.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	switch v.Kind() {
+	case cue.StructKind:
+		// If current cue.Value is a reference to another
+		// definition, we don't want to traverse its fields
+		// individually, because we'll do so for the actual def.
+		if v != cue.Dereference(v) {
+			return
+		}
```
Comment (joanlopez) [inline pkg/kindsys/kindsysreport/kindsysreport.go:44]: @sdboyer Here's what solves the duplicates, as discussed [here](https://github.com/grafana/schematization-and-as-code-project/issues/41#issuecomment-1398802441).

Comment (sdboyer) [reply pkg/kindsys/kindsysreport/kindsysreport.go:45]: Ah yeah, this is an even simpler way to solve it than what i suggested with a `map[cue.Value]bool`.

It also has the benefit that it won't escape the hierarchy of the schema declaration itself - like, if we have an `@grafanamaturity` attribute in the common schemas, it won't get counted in every kind that references it. (Though that may be something we want to do later 🤷🏻 )

LGTM!

File: pkg/kindsys/report.json [hunk 2]
```diff
@@ -223,6 +235,7 @@
       ],
       "docsRef": "https:/grafana.com/docs/grafana/next/developers/kinds/core/dashboard/schema-reference",
       "goRef": "https:/github.com/grafana/grafana/tree/main/pkg/kinds/dashboard/dashboard_kind_gen.go",
+      "grafanaMaturityCount": 144,
```
Comment (sdboyer) [inline pkg/kindsys/report.json:238]: roflcopter

**Review by sdboyer — APPROVED** [review]
fantastic!

**Pull request merged**